primitives/arithmetic/src/per_things.rs

```diff
@@ -71,48 +114,73 @@ macro_rules! implement_per_thing {
 			#[cfg(feature = "std")]
-			pub fn from_fraction(x: f64) -> Self { Self((x * ($max as f64)) as $type) }
+			fn from_fraction(x: f64) -> Self { Self((x * ($max as f64)) as Self::Inner) }
 
 			/// Approximate the fraction `p/q` into a per-thing fraction. This will never overflow.
 			///
 			/// The computation of this approximation is performed in the generic type `N`. Given
 			/// `M` as the data type that can hold the maximum value of this per-thing (e.g. u32 for
 			/// perbill), this can only work if `N == M` or `N: From<M> + TryInto<M>`.
-			pub fn from_rational_approximation<N>(p: N, q: N) -> Self
-				where N: Clone + Ord + From<$type> + TryInto<$type> + ops::Div<N, Output=N>
+			fn from_rational_approximation<N>(p: N, q: N) -> Self
+				where N: Clone + Ord + From<Self::Inner> + TryInto<Self::Inner> + ops::Div<N, Output=N>
 			{
 				// q cannot be zero.
-				let q = q.max((1 as $type).into());
+				let q = q.max((1 as Self::Inner).into());
 				// p should not be bigger than q.
 				let p = p.min(q.clone());
 
-				let factor = (q.clone() / $max.into()).max((1 as $type).into());
+				let factor = (q.clone() / $max.into()).max((1 as Self::Inner).into());
 
 				// q cannot overflow: (q / (q/$max)) < 2 * $max. p < q hence p also cannot overflow.
-				// this implies that $type must be able to fit 2 * $max.
-				let q_reduce: $type = (q / factor.clone())
+				// this implies that Self::Inner must be able to fit 2 * $max.
+				let q_reduce: Self::Inner = (q / factor.clone())
 					.try_into()
 					.map_err(|_| "Failed to convert")
 					.expect(
 						"q / (q/$max) < (2 * $max). Macro prevents any type being created that \
 						does not satisfy this; qed"
 					);
-				let p_reduce: $type = (p / factor.clone())
+				let p_reduce: Self::Inner = (p / factor.clone())
 					.try_into()
 					.map_err(|_| "Failed to convert")
 					.expect(
 						"q / (q/$max) < (2 * $max). Macro prevents any type being created that \
 						does not satisfy this; qed"
 					);
 
-				// `p_reduced` and `q_reduced` are withing $type. Mul by another $max will always
-				// fit in $upper_type. This is guaranteed by the macro tests.
+				// `p_reduced` and `q_reduced` are withing Self::Inner. Mul by another $max will
+				// always fit in $upper_type. This is guaranteed by the macro tests.
 				let part =
 					p_reduce as $upper_type
 					* <$upper_type>::from($max)
 					/ q_reduce as $upper_type;
 
-				$name(part as $type)
+				$name(part as Self::Inner)
+			}
+		}
+
+		/// Implement const functions
+		impl $name {
+			/// From an explicitly defined number of parts per maximum of the type.
+			///
+			/// This can be called at compile time.
+			pub const fn from_parts_const(parts: $type) -> Self {
+				Self([parts, $max][(parts > $max) as usize])
+			}
+
+			/// Converts a percent into `Self`. Equal to `x / 100`.
+			///
+			/// This can be created at compile time.
+			pub const fn from_percent_const(x: $type) -> Self {
```

Comment:
Why not just call it `from_percent`? The compiler should be able to pick the correct one.